feat: add `s.string.phone`

imranbarbhuiya opened the pull request:
fixes: #200

vladfrangu commented on tests/validators/string.test.ts:
```diff
@@ -365,31 +365,51 @@ describe('StringValidator', () => {
+				}
+			);
+
+			test.each(['+1 555-555-5555 ext', '+1 (555) 555-5555', '987-123-4567 x12345', '(123) 456-7890 ext12345'])(
```
Why does `+1 (555) 555-5555` fail exactly? it *is* a valid number 🤔

imranbarbhuiya replied:
I used the regex fava provided. I don't have much idea of numbers with area code. Perhaps u can't use area code with country code? If that's not the case, I'll search for a better regex

favna replied:
looks like this one may be better:
```ts
^(\+\d{1,2}\s)?\(?\d{3}\)?[\s.-]?\d{3}[\s.-]?\d{4}$
```

Source: https://stackoverflow.com/a/16699507

imranbarbhuiya commented:
I've updated the regex. Also changed `phone` from `getter` to a `method`. The reason is maybe we can add `allowedCountryCode`. Will try to do that in a followup pr

vladfrangu commented:
Getting that quota are we? :P Yeah, sure a different PR for adding that would be fine (since it'd also need more testing)

favna reviewed (approved):
sounds quite complex to do country code so you rather than me lol. But hey if you can figure it out then awesome.
